packages/expo-file-system/ios/Next/FileSystemFile.swift

```diff
@@ -2,9 +2,24 @@ import Foundation
 import ExpoModulesCore
 
 internal final class FileSystemFile: FileSystemPath {
+  override init(url: URL) {
+    if !url.hasDirectoryPath {
+      super.init(url: url)
+      return
+    }
+    super.init(url: url.deletingLastPathComponent().appendingPathComponent(url.lastPathComponent, isDirectory: false))
```

Comment:
Same here. Maybe the constructor can be moved to `FileSystemPath` as `init(url: URL, isDirectory: Bool)` so we don't have to duplicate the code?